### PR TITLE
Fix image version handling for v1.2.0-rc1

### LINKED ISSUE #1797: docker image "v1.2.0-rc1" breaks ansible docker version parsing.
Hi.

# docker run --rm openshift/origin:latest version
openshift v1.2.0-rc1-17-g642f0af
kubernetes v1.2.0-36-g4a3f9c5
etcd 2.2.5

Issue is related roles/openshift_docker/tasks/main.yml:

    l_image_tag: "{{ cli_image_version.stdout_lines[0].split(' ')[1].split('-')[0] }}"

when running with the previous result, it leads to "v1.2.0" and not "v1.2.0-rc1", so it breaks the version tag selection on docker images.

### LINKED ISSUE #1800: Tag v1.2.0 not found in repository docker.io/openshift/origin
openshift-ansible fails on centos atomic host (containerized deployment) when installing openshift origin. It seems that openshift_version/image_tag is evaluated to v1.2.0 but only v1.2.0-rc1 tag exists on docker.io.

TASK: [openshift_cli | Install clients] *************************************** 
skipping: [dns1-origin-master-0.example.com]

TASK: [openshift_cli | Pull CLI Image] **************************************** 
<dns1-origin-master-0.example.com> ESTABLISH CONNECTION FOR USER: centos
<dns1-origin-master-0.example.com> REMOTE_MODULE command docker pull openshift/origin:v1.2.0
<dns1-origin-master-0.example.com> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/root/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 dns1-origin-master-0.example.com /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1461584253.73-95265263053286 && echo $HOME/.ansible/tmp/ansible-tmp-1461584253.73-95265263053286'
<dns1-origin-master-0.example.com> PUT /tmp/tmpG25r2K TO /home/centos/.ansible/tmp/ansible-tmp-1461584253.73-95265263053286/command
<dns1-origin-master-0.example.com> EXEC ssh -C -tt -vvv -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/root/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=10 dns1-origin-master-0.example.com /bin/sh -c 'sudo -k && sudo -H -S -p "[sudo via ansible, key=bsaysusaqtmcirmxftjtdagoifwxolpp] password: " -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-bsaysusaqtmcirmxftjtdagoifwxolpp; LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /home/centos/.ansible/tmp/ansible-tmp-1461584253.73-95265263053286/command; rm -rf /home/centos/.ansible/tmp/ansible-tmp-1461584253.73-95265263053286/ >/dev/null 2>&1'"'"''
failed: [dns1-origin-master-0.example.com] => {"changed": true, "cmd": ["docker", "pull", "openshift/origin:v1.2.0"], "delta": "0:00:02.840508", "end": "2016-04-25 11:37:36.921582", "rc": 1, "start": "2016-04-25 11:37:34.081074", "warnings": []}
stderr: Tag v1.2.0 not found in repository docker.io/openshift/origin
stdout: Trying to pull repository docker.io/openshift/origin ... not found

FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/root/main.retry

dns1-infra.example.com     : ok=14   changed=10   unreachable=0    failed=0   
dns1-origin-master-0.example.com : ok=291  changed=54   unreachable=0    failed=1   
dns1-origin-node-hzr604rn.example.com : ok=15   changed=1    unreachable=0    failed=0   
localhost                  : ok=20   changed=0    unreachable=0    failed=0

### PR DESCRIPTION
**Pull request by sdodson**
Fixes #1800
Fixes #1797

Resultant openshift.docker facts
```
"docker": {
...
    "openshift_image_tag": "v1.2.0-rc1",
    "openshift_version": "v1.2.0"
},
```

### CODE REVIEW

**Comment by sdodson** [comment]
aos-ci-test

**Comment by sdodson** [comment]
Currently fails on OSE, perhaps only include second component when deployment_type == origin?
```
TASK: [openshift_cli | Pull CLI Image] **************************************** 
failed: [10.8.170.211] => {"changed": true, "cmd": ["docker", "pull", "openshift3/ose:v3.1.1.6-33"], "delta": "0:00:02.031128", "end": "2016-04-25 13:24:36.713101", "rc": 1, "start": "2016-04-25 13:24:34.681973", "warnings": []}
stderr: Error: image openshift3/ose:v3.1.1.6-33 not found
stdout: Trying to pull repository XXXXXXXXXXXXXXXXXXXXXXXXXXX/openshift3/ose ... not found
Trying to pull repository docker.io/openshift3/ose ... not found
```

**Comment by sdodson** [comment]
aos-ci-test

**Comment by sdodson** [comment]
@brenton PTAL, not a blocker for 3.2 but a hassle for origin containerized installs

**Comment by sdodson** [comment]
aos-ci-test

File: roles/openshift_docker/tasks/main.yml [hunk 1]
```diff
@@ -9,7 +9,8 @@
   when: openshift.common.is_containerized is defined and openshift.common.is_containerized | bool and openshift_image_tag is not defined
 
 - set_fact:
-    l_image_tag: "{{ cli_image_version.stdout_lines[0].split(' ')[1].split('-')[0] }}"
+    l_image_tag: "{{ cli_image_version.stdout_lines[0].split(' ')[1].split('-')[0:2] | join('-') if openshift.common.deployment_type == 'origin' else
```
Comment (detiber) [inline roles/openshift_docker/tasks/main.yml]: This seems a bit error prone

Comment (sdodson) [inline roles/openshift_docker/tasks/main.yml]: In general or after my change? I think we only need concern ourselves with Origin here, OSE variants are so coupled to RPM build process that I cannot foresee us ever relying on a second version component mapping to image versions.

So, for origin I looked at 'latest', 'v1.1.6', and 'v1.2.0-rc1' and this should work for them all.

**Pull request merged**